patch-test.js

```diff
@@ -0,0 +1,86 @@
+/* eslint-env node, es6 */
+/* eslint-disable no-console */
+
+/**
+ * Creates a file patch-test-loader.js with the necessary import statements
+ * to test the changes made in a pull request or in the current working state
+ *
+ * How to use:
+ * - Run `npm run patchtest` which generates patch-test-loader.js
+ * - Set up a localhost server (such as by using server.js) and load patch-test-loader.js
+ *   in the wiki environment, using the browser console or the common.js page.
+ */
+
+const fs = require('fs');
+const {exec} = require('child_process');
+
+// find the last common commit between this branch and master, and get the list
+// of files changed since that commit
+exec('git diff $(git merge-base $(git rev-parse --abbrev-ref HEAD) master) --name-only', (err, stdout) => {
```

Comment:
Breaking this down:
- `git rev-parse --abbrev-ref HEAD` gives the name of current branch
- `git merge-base brName master` gives the last common commit b/w this branch and master
- `git diff commitID` gives the diff of that commit with the current working directory (changes from subsequent commits, and any staged and unstaged changes). With the `--name-only` flag it just lists names of files changed.